packages/@uppy/companion/src/server/provider/Provider.js

```diff
@@ -64,17 +68,22 @@ class Provider {
    * @param {object} options
    * @returns {Promise}
    */
-  async deauthorizationCallback (options) { // eslint-disable-line no-unused-vars
+  // eslint-disable-next-line class-methods-use-this,no-unused-vars
+  async deauthorizationCallback (options) {
     // @todo consider doing something like throw new NotImplementedError() instead
     throw new Error('method not implemented')
   }
 
   /**
+   * Name of the OAuth provider. Return empty string if no OAuth provider is needed.
+   *
    * @returns {string}
    */
   static get authProvider () {
-    return ''
+    return undefined
   }
 }
 
 module.exports = Provider
+// OAuth providers are those that have a `static authProvider` set. It means they require OAuth authentication to work
+module.exports.isOAuthProvider = (authProvider) => authProvider != null
```

Comment:
```suggestion
module.exports.isOAuthProvider = (authProvider) => authProvider !== null
```

Reply:
who do we want a missing `authProvider` property to count as having an authProvider?
What if someone creates a custom provider, and the implement authProvider as returning undefined, then it will be treated as an authprovider if we make this change. I think it's nice to treat both null and undefined as *no auth provider*

Reply:
In that case why not check whether it's a string and the string is longer than 0?
- `isOAuthProvider` implies a validity check, not invalidity
- is more strict, passing random things like boolean, objects, array, etc will fail.

Reply:
yes, that's an even better check!